recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,170 @@
+            self.requires("libpq/12.2")
+
+        if self.options.with_boost:
+            self.requires("boost/1.72.0")
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        url = self.conan_data["sources"][self.version]["url"]
+        archive_name = os.path.basename(url)
+        archive_name = os.path.splitext(archive_name)[0]
+        os.rename("soci-%s" % archive_name, self._source_subfolder)
+        
+    def _dependency(self, name):
+        def find_lib_path(name, lib):
+            lib_filename = (
+              '' if self.settings.os == "Windows" else 'lib'
+            ) + lib + (
+              '.lib' if self.settings.os == "Windows" else '.so' if self.options[name].shared else '.a'
+            )
+
+            for path in self.deps_cpp_info[name].lib_paths:
+                full_path = os.path.join(
+                  path,
+                  lib_filename
+                )
+                if os.path.isfile(full_path):
+                    return full_path
+            print('{} not found in {}'.format(lib_filename, self.deps_cpp_info[name].lib_paths))
+            return None
+
+        libs = [find_lib_path(name, lib) for lib in self.deps_cpp_info[name].libs]
+
+        return namedtuple('Dependency', ['root', 'libs'])(
+          root          = self.deps_cpp_info[name].rootpath,
+          libs          = libs
+        )
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["SOCI_SHARED"] = self.options.shared
+        cmake.definitions["SOCI_STATIC"] = not self.options.shared
+        cmake.definitions["SOCI_TESTS"] = False
+
+        print("self.settings.compiler.cppstd = {}".format(self.settings.compiler.cppstd))
+        if self.settings.compiler.cppstd != 'None':
+            cmake.definitions["CMAKE_CXX_STANDARD"] = self.settings.compiler.cppstd
+
+        cmake.definitions["WITH_EMPTY"] = self._with_backend(self.options.with_backend_empty)
+
+        if self._with_backend(self.options.with_backend_sqlite3):
+            sqlite3 = self._dependency('sqlite3')
+            cmake.definitions["WITH_SQLITE3"] = "ON"
+            cmake.definitions["SQLITE_ROOT_DIR"] = sqlite3.root
+
+        if self._with_backend(self.options.with_backend_mysql):
+            mysql = self._dependency('libmysqlclient')
+            cmake.definitions["WITH_MYSQL"] = "ON"
+            os.environ["MYSQL_DIR"] = mysql.root
+            cmake.definitions["MYSQL_LIBRARIES"] = ';'.join(mysql.libs)
+
+        if self._with_backend(self.options.with_backend_odbc):
+            odbc = self._dependency('odbc')
+            cmake.definitions["WITH_ODBC"] = "ON"
+            cmake.definitions["ODBC_INCLUDE_DIR"] = os.path.join(odbc.root, 'include')
+            if self.settings.os == 'Windows':
+              cmake.definitions["ODBC_LIBRARY"] = os.path.join(odbc.root, 'lib', 'libodbc.lib')
```

Comment:
Doesn't this work?
```suggestion
              cmake.definitions["ODBC_LIBRARY"] = ";".join(seld.deps_cpp_info["odb"].libs)
```
It the linker complains about not finding the libraries, you can add `link_libraries(${CONAN_LIB_DIRS})` to your `CMakeLists.txt`.

Reply:
Also, hardcoding the `.lib` extension will not work when using mingw.